In-App: make in-app header sticky

raicem opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

This PR makes the in-app header sticky so that it's visible at all times. The goal is to improve discoverability. 

Closes https://github.com/Automattic/woocommerce.com/issues/21566.

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

1. Checkout this branch and get a new build
2. Load [the in-app marketplace](http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=extensions&path=%2Fextensions) and scroll down. Confirm the header is now sticky. 
3. Interact with the header: click all the tabs and scroll down. Click on the user avatar and open menu. Connect and disconnect to WCCOM. 
4. Check if you see a double scrollbar, like show in the screenshot from PR [Suppress second scrollbar on WooCommerce > Extensions pages #40046](https://github.com/woocommerce/woocommerce/pull/40046). 
    - I had to delete that fix, to make `position: sticky` work. After deleting it, I didn't get a second scrollbar. Could be a platform difference (I'm on Linux + Firefox). So curious to see what you'll see. 
6. Please test around this issue.

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->
Make In-App Marketplace header sticky to improve discoverability. 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

### Screenshots
![image](https://github.com/user-attachments/assets/271999d2-f4db-40b5-96c1-34a577a98b61)

raicem committed:
In-App: make in-app header sticky

z-index: 1 is required to make sure the header appears above product
cards.

Also had to remove a parent `overflow: hidden` property. Apparently,
it's there to remove a second scrollbar. Upon removal, I didn't see
any double scrollbars, although it can be platform differences. Will
check it out in the PR review.

raicem commented on plugins/woocommerce-admin/client/marketplace/components/header/header.scss:
```diff
@@ -9,6 +9,9 @@
 	width: 100%;
+	position: sticky;
+	top: var(--wp-admin--admin-bar--height, 32px);
+	z-index: 1;
```
I chose the minimum possible z-index here, although I can understand an argument for setting it to 10 or 100.

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

raicem commented on plugins/woocommerce-admin/client/marketplace/marketplace.scss:
```diff
@@ -7,11 +7,6 @@
 
-	#wpbody-content {
-		/* Prevent double-scrollbar issue on WooCommerce > Extension pages */
-		overflow: hidden;
```
This is important. From testing instructions:

>  I had to delete that fix, to make position: sticky work. After deleting it, I didn't get a second scrollbar. Could be a platform difference (I'm on Linux + Firefox). So curious to see what you'll see.

KokkieH reviewed (commented):
Works great on desktop, but on mobile I'm seeing some weird behaviour on some tabs: On any tab that shows theme cards, so Themes, Discover, and sometimes Search, the header isn't properly sticky, but scrolls up with the content, then reappears as sticky when you scroll down.

https://github.com/user-attachments/assets/0b675287-28a0-4d70-9657-c4cdd9a4f278

I also noticed that the theme cards appear wider than the viewport on mobile. It's happening on `trunk` as well, so not related to your changes, but I wonder if it might be related to what's causing the weird scroll behaviour. If it's unrelated, it should still be fixed in a separate task.

raicem committed:
Update Change Your Store width for Marketplace search listing

Setting it 450px hardcoded causes overflows and breaks the layout.

See:
https://github.com/woocommerce/woocommerce/pull/51187#issuecomment-2344003169

raicem commented:
A hardcoded value for change your store banner was breaking the layout. [Updated](https://github.com/woocommerce/woocommerce/pull/51187/commits/5e840f924b2bd0c35dad0b6854b97f720927d156) it for Marketplace pages. 

![image](https://github.com/user-attachments/assets/e9ef70a6-8f72-477d-936f-efc3a97bac94)

KokkieH commented on plugins/woocommerce-admin/client/marketplace/components/content/content.scss:
```diff
@@ -8,6 +8,10 @@
 	min-height: 500px;
 }
 
+.woocommerce-marketplace__content .woocommerce-customize-store-banner .woocommerce-customize-store-banner-content {
```
Can we maybe move this style to `plugins/woocommerce-admin/client/customize-store/intro/intro.scss` instead? It looks like most of the banner's styles are defined there, and it makes more sense to me to keep those in one place.

In the same vein there's a single style for the banner defined in `plugins/woocommerce-admin/client/marketplace/components/product-list-content/product-list-content.scss`, so maybe we can move that as well?

raicem replied:
Sure thing, I can move it. I approached it this way: Customize Your Store is a reusable component but this tweak we're making is special to our page. Probably we'll maintain this placement and make sure it looks good in future iterations of the marketplace. So I put in our code but I understand your suggestion too

KokkieH replied:
Okay, if this is marketplace-specific it makes sense to keep it here. In that case I wouldn't duplicate it in intro.scss. As it is, the current code is causing a build error, as the `$breakpoint-medium` var isn't available in that file. I've taken the liberty to remove those styles 0e8e577 so I can build and test the margin change :)

raicem replied:
Whoops, so sorry. I moved it but forgot to remove the old rules. Thank you so much

KokkieH reviewed (commented):
Great catch! That looks much better.

I do notice the banner text now overlaps with the background image:

![Cursor_and_Extensions_‹_WooCommerce_‹_WooCommerce_Core_E2E_Test_Suite](https://github.com/user-attachments/assets/04c6af40-c58c-4d29-a95f-6f4d5352b281)

Reducing the margin-top on the `__content` class for smaller screens should work here (that margin is currently defined in `product-list-content.scss`, but I think that style should be moved to the rest of the banner styles. See comment in code.

raicem committed:
Move CYS tweak to the component

It was placed in the marketplace SCSS code previously. GitHub:
https://github.com/woocommerce/woocommerce/pull/51187#discussion_r1758381473

KokkieH committed:
Merge branch 'trunk' into update/21566-in-app-marketaplace-header-sticky

KokkieH committed:
Remove marketplace-specific styles from customize store banner

KokkieH reviewed (approved):
Thanks for the changes, Cem. I think this is good to go.

raicem committed:
Merge branch 'trunk' into update/21566-in-app-marketaplace-header-sticky
